src/models/room.ts

```diff
@@ -933,6 +1051,10 @@ export class Room extends TypedEventEmitter<EmittedEvents, RoomEventHandlerMap>
             .getState(EventTimeline.BACKWARDS);
         this.currentState = this.getLiveTimeline()
             .getState(EventTimeline.FORWARDS);
+
+        // Let people know to register listeners for the new state references
+        this.emit(RoomEvent.OldStateUpdated, this, previousOldState, this.currentState);
```

Comment:
`RoomEvent.OldStateUpdated` is not used but seemed applicable to add the equivalent to `RoomEvent.CurrentStateUpdated` since people could be doing the same thing with `oldState` for whatever reason. Also willing to remove since we aren't exercising it ourselves.

Reply:
Re-commented on latest diff: https://github.com/matrix-org/matrix-js-sdk/pull/2299/files#r883240213